src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DriverDecorator.java

```diff
@@ -1,16 +1,15 @@
 package edu.kis.powp.jobs2d.drivers.adapter;
 
-import edu.kis.powp.jobs2d.Job2dDriver;
 import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
 import edu.kis.powp.jobs2d.drivers.visitor.IDriverVisitor;
 
 public interface DriverDecorator extends VisitableJob2dDriver {
-    Job2dDriver getDriver();
+    VisitableJob2dDriver getDriver();
 
-    void setDriver(Job2dDriver driver);
+    void setDriver(VisitableJob2dDriver driver);
 
     @Override
-    default void accept(IDriverVisitor visitor) {
-        visitor.visit(this);
+    default long accept(IDriverVisitor visitor) {
```

Comment:
sorry, but accept should not return any particular type of value. Instead of returning a client after calling accept can simply check the result using a getter (specified in the given visitor).